torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,1063 @@
+
+        calibrartion_suffixes = {
+            None: [""],
+            "perfect": ["-perfect"],
+            "imperfect": ["-imperfect"],
+            "both": ["-perfect", "-imperfect"],
+        }[calibration]
+
+        for calibration_suffix in calibrartion_suffixes:
+            scene_pattern = "*" + calibration_suffix
+
+            imgs_left = sorted(glob(str(root / split / scene_pattern / "im0.png")))
+            imgs_right = sorted(glob(str(root / split / scene_pattern / "im1.png")))
+            if not len(imgs_left) or not len(imgs_right):
+                raise FileNotFoundError("No images found in {}".format(root))
+
+            self._images += list((left, right) for left, right in zip(imgs_left, imgs_right))
+
+            if split == "test":
+                disparity_maps_left, disparity_maps_right = list("" for _ in imgs_left), list("" for _ in imgs_right)
+            else:
+                disparity_maps_left = sorted(glob(str(root / split / "*" / "disp0.pfm")))
+                disparity_maps_right = sorted(glob(str(root / split / "*" / "disp1.pfm")))
+                if not len(disparity_maps_left) or not len(disparity_maps_right):
+                    raise FileNotFoundError("No disparity maps found in {}".format(root / split))
+
+            self._disparities += list((left, right) for left, right in zip(disparity_maps_left, disparity_maps_right))
+
+        self.use_ambient_views = use_ambient_views
+
+    def __getitem__(self, index: int) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+        return super().__getitem__(index)
+
+    def _read_img(self, file_path: str) -> Image.Image:
+        """Function that reads either the original right image or an augmented view when ``use_ambient_views`` is True."""
+        if os.path.basename(file_path) == "im1.png" and self.use_ambient_views:
+            # initialize sampleable container
+            ambient_file_paths = list(file_path.replace("im1.png", view_name) for view_name in ["im1E.png", "im1L.png"])
+            # double check that we're not going to try to read from an invalid file path
+            ambient_file_paths = list(filter(lambda p: os.path.exists(p), ambient_file_paths))
+            # keep the original image as an option as well for uniform sampling between base views
+            ambient_file_paths.append(file_path)
+            file_path = random.choice(ambient_file_paths)
+        return super()._read_img(file_path)
+
+    def _read_disparity(self, file_path: str) -> Tuple:
+        if not os.path.exists(file_path):  # case when dealing with the test split
+            return None, None
+        disparity_map = read_pfm_file(file_path)
+        valid_mask = disparity_map < 1e3
+        # remove the channel dimension from the valid mask
+        valid_mask = valid_mask[0, :, :]
+        return disparity_map, valid_mask
+
+    def _download_dataset(self, root: str):
```

Comment:
I've left downloads just to check what we want to do further down the line. Since the CRE dataset is about 400 GB I was talking with @jdsgomes  about adopting the Imagenet route, and not provide a download method.

Just looking to get some final feedback on how we should approach this. For Middlebury it's definitely more than possible to do that.